Yet another flaky UI test: UC112_T01_UI_toc

```py
            ...
            # toc is hidden, but still contains Section title

            screen_toc.assert_toc_closed()

            screen_toc.assert_toc_contains("Section title")

            # show the toc

            screen_toc.do_toggle_toc()

            # toc is open
>           screen_toc.assert_toc_opened()

tests\end2end\navigation\toc\test_UC112_T01_UI_toc.py:48: 
```

```
exception = <class 'selenium.common.exceptions.NoSuchElementException'>

message = 'Element {//*[@data-testid="toc-bar"][@data-state="open"]} was not present after 10 seconds!'



    def timeout_exception(exception, message):

        exc, msg = shared_utils.format_exc(exception, message)

>       raise exc(msg)

E       selenium.common.exceptions.NoSuchElementException: Message: 

E        Element {//*[@data-testid="toc-bar"][@data-state="open"]} was not present after 10 seconds!
```

--- a/.github/workflows/periodic-release-test.yml
+++ b/.github/workflows/periodic-release-test.yml
@@ -29,7 +29,7 @@ jobs:
 
       - name: Build and test locally
         run: |
-          pip install strictdoc
+          pip install strictdoc --pre
 
       - name: Integration tests against the StrictDoc Pip package
         run: |

--- a/strictdoc/core/document_finder.py
+++ b/strictdoc/core/document_finder.py
@@ -70,15 +70,12 @@ class DocumentFinder:
             map_docs_by_paths[input_doc_full_path] = document
             document_list.append(document)
 
-        for file_tree, doc_file, file_tree_mount_folder in file_tree_list:
+        for _, doc_file, file_tree_mount_folder in file_tree_list:
             input_doc_full_path = doc_file.get_full_path()
             document = map_docs_by_paths[input_doc_full_path]
             assert isinstance(document, Document)
 
-            doc_relative_path = os.path.relpath(
-                input_doc_full_path, file_tree.root_path
-            )
-            doc_relative_path_folder = os.path.dirname(doc_relative_path)
+            doc_relative_path_folder = os.path.dirname(doc_file.rel_path)
 
             output_document_dir_rel_path = (
                 f"{file_tree_mount_folder}/{doc_relative_path_folder}"

--- a/strictdoc/core/file_tree.py
+++ b/strictdoc/core/file_tree.py
@@ -20,12 +20,14 @@ class FileOrFolderEntry:
 
 
 class File(FileOrFolderEntry):
-    def __init__(self, level, full_path):
+    def __init__(self, level, full_path, rel_path: str):
         assert os.path.isfile(full_path)
         assert os.path.isabs(full_path)
+        assert isinstance(rel_path, str)
         self.level = level
         self.full_path = full_path
         self.root_path = full_path
+        self.rel_path = rel_path
         self.files = [self]
         self.subfolder_trees = []
 
@@ -84,7 +86,10 @@ class Folder(FileOrFolderEntry):  # pylint: disable=too-many-instance-attributes
     def set(self, files):
         for file in files:
             full_file_path = os.path.join(self.root_path, file)
-            self.files.append(File(self.level + 1, full_file_path))
+            rel_file_path = os.path.join(self.rel_path, file)
+            self.files.append(
+                File(self.level + 1, full_file_path, rel_file_path)
+            )
 
     def add_subfolder_tree(self, subfolder_tree):
         assert isinstance(subfolder_tree, Folder)
@@ -101,7 +106,7 @@ class FileTree:
 
     @staticmethod
     def create_single_file_tree(root_path):
-        single_file = File(0, root_path)
+        single_file = File(0, root_path, "")
         return FileTree(root_folder_or_file=single_file)
 
     def iterate(self):

--- a/tests/end2end/navigation/collapsible_list/test_UC114_T01_UI_collapsible_list.py
+++ b/tests/end2end/navigation/collapsible_list/test_UC114_T01_UI_collapsible_list.py
@@ -8,7 +8,7 @@ from tests.end2end.helpers.screens.project_index.screen_project_index import (
 from tests.end2end.server import SDocTestServer
 
 
-class Test_UC112_T01_UI_toc(BaseCase):
+class Test_UC114_T01_UI_collapsible_list(BaseCase):
     def test_01(self):
         test_setup = End2EndTestSetup(path_to_test_file=__file__)
 

--- a/tests/end2end/navigation/toc/test_UC112_T01_UI_toc.py
+++ b/tests/end2end/navigation/toc/test_UC112_T01_UI_toc.py
@@ -41,6 +41,13 @@ class Test_UC112_T01_UI_toc(BaseCase):
             screen_toc.assert_toc_closed()
             screen_toc.assert_toc_contains("Section title")
 
+            # Suspecting that this is a reason for the flaky test:
+            # Element {//*[@data-testid="toc-bar"][@data-state="open"]} was not
+            # present after 7 seconds!
+            # https://github.com/strictdoc-project/strictdoc/issues/1155
+            # Wait for animation to complete.
+            self.sleep(1)
+
             # show the toc
             screen_toc.do_toggle_toc()
 

